flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/sort/SortingBacklogDataInputTest.java

```diff
@@ -0,0 +1,235 @@
+                                new StreamRecord<>(1, 3),
+                                new Watermark(1),
+                                new RecordAttributesBuilder(Collections.emptyList())
+                                        .setBacklog(true)
+                                        .build(),
+                                new StreamRecord<>(1, 1),
+                                new Watermark(2),
+                                new StreamRecord<>(2, 1),
+                                new Watermark(3),
+                                new StreamRecord<>(2, 3),
+                                new Watermark(4),
+                                new StreamRecord<>(1, 2),
+                                new Watermark(5),
+                                new StreamRecord<>(2, 2),
+                                new Watermark(6),
+                                new RecordAttributesBuilder(Collections.emptyList())
+                                        .setBacklog(false)
+                                        .build()));
+        MockEnvironment environment = MockEnvironment.builder().build();
+        SortingBacklogDataInput<Integer, Integer> sortingDataInput =
+                new SortingBacklogDataInput<>(
+                        input,
+                        new IntSerializer(),
+                        new IntSerializer(),
+                        (KeySelector<Integer, Integer>) value -> value,
+                        environment.getMemoryManager(),
+                        environment.getIOManager(),
+                        true,
+                        1.0,
+                        new Configuration(),
+                        new DummyInvokable(),
+                        new ExecutionConfig(),
+                        () -> true);
+
+        DataInputStatus inputStatus;
+        do {
+            inputStatus = sortingDataInput.emitNext(collectingDataOutput);
+        } while (inputStatus != DataInputStatus.END_OF_INPUT);
+
+        org.assertj.core.api.Assertions.assertThat(collectingDataOutput.events)
+                .containsExactly(
+                        new StreamRecord<>(1, 3),
+                        new Watermark(1),
+                        new RecordAttributesBuilder(Collections.emptyList())
+                                .setBacklog(true)
+                                .build(),
+                        new InternalRecordAttributes(true, 6L),
+                        new StreamRecord<>(1, 1),
+                        new StreamRecord<>(1, 2),
+                        new StreamRecord<>(2, 1),
+                        new StreamRecord<>(2, 2),
+                        new StreamRecord<>(2, 3),
+                        new RecordAttributesBuilder(Collections.emptyList())
+                                .setBacklog(false)
+                                .build(),
+                        new Watermark(6));
+    }
+
+    @Test
+    public void simpleVariableLengthKeySorting() throws Exception {
```

Comment:
What does this test case verify in addition to other test cases?

Reply:
This test the variable length key similar to the test in `SortingDataInputTest`.